### ISSUE TITLE
When a file has its date set in frontmatter ContentType breaks

### ISSUE THREAD

**Issue by carbontwelve** [carbontwelve/tapestry#89]
When a file has date set in its frontmatter e.g:
```
---
title: test
date: 11/09/2016
---
```

or
```
---
title: test
date: 1451606400
---
```

Then it's not converted to an instance of `DateTime` which is what line 119 of ContentType is expecting. Expected behavior is that the date is always an instance of `DateTime` with front matter input being injected.

In the future the frontmatter date property will be used for scheduling posts to become published at a certain time and so should accept either a date in the format DD/MM/YYYY or as a unix timestamp.

**Comment by carbontwelve** [comment]
This can be fixed by adding the following to the `setData` method of the File entity.

```
if (isset($data['date']) && !($data['date'] instanceof DateTime)) {
            $date = new DateTime();
            $date->createFromFormat('U', strtotime($data['date']));
            $data['date'] =  $date;
        }
```

**Comment by carbontwelve** [comment]
fixed by 107899be157fbb63a33cab7a1c8d3ef26b04fba5
